src/Idna/IdnaConverter.php

```diff
@@ -0,0 +1,87 @@
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * 	* Redistributions of source code must retain the above copyright notice, this list of
+ * 	  conditions and the following disclaimer.
+ *
+ * 	* Redistributions in binary form must reproduce the above copyright notice, this list
+ * 	  of conditions and the following disclaimer in the documentation and/or other materials
+ * 	  provided with the distribution.
+ *
+ * 	* Neither the name of the SimplePie Team nor the names of its contributors may be used
+ * 	  to endorse or promote products derived from this software without specific prior
+ * 	  written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS
+ * AND CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @copyright 2004-2022 Ryan Parman, Sam Sneddon, Ryan McCue
+ * @author Ryan Parman
+ * @author Sam Sneddon
+ * @author Ryan McCue
+ * @link http://simplepie.org/ SimplePie
+ * @license http://www.opensource.org/licenses/bsd-license.php BSD License
+ */
+
+namespace SimplePie\Idna;
+
+use Algo26\IdnaConvert\IdnaConvert;
+use Algo26\IdnaConvert\ToUnicode;
+use idna_convert;
+
+/**
+ * Convert internationalized domain names using idna-convert
+ *
+ * @internal
+ */
+final class IdnaConverter implements IdnaDomainFilter
+{
+    /**
+     * Convert an internationalized domain name.
+     *
+     * e.g. from `xn--mller-kva` to `müller`
+     *
+     * @param string $encoded The encoded domain name
+     *
+     * @return string the decoded domain name
+     */
+    public function filterDomain(string $encoded): string
+    {
+        if (class_exists(ToUnicode::class)) {
+            // Support for algo26-matthias/idna-convert:^3
+            $idnaConvert = new ToUnicode();
+
+            return $idnaConvert->convert($encoded);
+        } else if (class_exists(IdnaConvert::class)) {
+            // Support for algo26-matthias/idna-convert:^2
+            $idnaConvert = new IdnaConvert();
+
+            return $idnaConvert->decode($encoded);
+        } else if (class_exists(idna_convert::class)) {
+            // Support for idna_convert:0.5.1
+            $idnaConvert = new idna_convert();
+
+            return $idnaConvert->decode($encoded);
```

Comment:
But here you are switching it to decoding.

Reply:
ups... 😅

Reply:
Fixed, see https://github.com/simplepie/simplepie/pull/776#discussion_r1092947362